File: codingsample/202008/20200824/Rains_of_Reason_2_evenDigitsOnly.java
package N6_Rains_of_Reason;

import java.util.Arrays;

public class Rains_of_Reason_2_evenDigitsOnly {
	static boolean evenDigitsOnly(int n) {
		int temp;
		while(n!=0) {
			temp = n;
			if(temp%2!=0) return false;
			n/=10;
		}
		return true;
	}
	static boolean evenDigitsOnly2(int n) {
		String temp = String.valueOf(n);
		for(char s : temp.toCharArray())  {
			if((s-'0') % 2 != 0) return false;
		}
		return true;
	}
	
	static boolean evenDigitsOnly3(int n) {
	    return Arrays.stream(Integer.toString(n).split("")).allMatch(i -> Integer.parseInt(i) % 2 ==0);}
	
	static boolean evenDigitsOnly4(int n) {
	    return Integer.toString(n).matches("(0|2|4|6|8)*");}
	
	static boolean evenDigitsOnly5(int n) {
	        return (n + "").matches("[02468]+");}

	public static void main(String[] args) {
		int a = 248622;
		int b = 42608;
		System.out.println((b+"").matches("(0|2|4|6|8)"));
	}

}

File: ~2020.11.26 - codingsample/202008/20200824/Rains_of_Reason_2_evenDigitsOnly.py
def evenDigitsOnly(n):
    while(n!=0):
        temp = int(n % 10)
        if temp % 2 != 0 : return False
        n = int(n/10)
    return True

def evenDigitsOnly2(n):
    temp = str(n)
    for i in temp:
        if int(i) % 2 != 0: return False
    return True
    
def evenDigitsOnly(n):
    return all([int(i)%2==0 for i in str(n)])

a = 248622
b = 642386
print(evenDigitsOnly2(a))
print(evenDigitsOnly2(b))